api/src/services/block-cursor.ts

```diff
@@ -10,6 +10,7 @@ class BlockCursorService {
     });
     if (!blockCursor) {
       blockCursor = await BlockCursor.query().insertAndFetch({
+        id: v4(),
```

Comment:
@10thfloor What's the need for the UUID here?

Reply:
SQlite does not have a native UUID type (unlike Postgres), so I have to provide a UUID here from JS.

UUIDs are needed because events from the contract are all the same, so we need a way to differentiate in the DB. I suppose we could use a timestamp as well?

Reply:
Got it. But this is for the block cursor, which I assume should be unique based on block height.

Reply:
Right. Hmm. The ID is stable, but the block height is updated each time we see an event? I suppoed we could use 
`blockCursor.currentBlockHeight` to query instead?

```
blockCursor = await this.blockCursorService.updateBlockCursorById(
                blockCursor.id,
                toBlock
              );
```